### PR TITLE
Narrow profiling scope

### PR DESCRIPTION
**Pull request by RandySheriffH**
Collecting op end time earlier to avoid noise generated by thread pool profiler.

### CODE REVIEW

File: onnxruntime/core/common/profiler.h [hunk 1]
```diff
@@ -78,6 +78,17 @@ class Profiler {
                              const std::initializer_list<std::pair<std::string, std::string>>& event_args = {},
                              bool sync_gpu = false);
 
+  void EndTimeAndRecordEvent(EventCategory category,
```
Comment (pranavsharma) [inline onnxruntime/core/common/profiler.h:81]: Do these new methods need to be public?

File: onnxruntime/core/common/profiler.h [hunk 2]
```diff
@@ -78,6 +78,17 @@ class Profiler {
...
+                             const std::string& event_name,
+                             const TimePoint& start_time, const TimePoint& end_time,
+                             const std::initializer_list<std::pair<std::string, std::string>>& event_args = {},
+                             bool sync_gpu = false);
+
+  void EndTimeAndRecordEvent(EventCategory category,
+                             const std::string& event_name,
+                             long long dur, long long ts,
```
Comment (pranavsharma) [inline onnxruntime/core/common/profiler.h:89]: What do 'dur' and 'ts' represent here? Both are of type 'long long' and it's easy to get confused and mess up things. Needs clear documentation.

File: onnxruntime/core/framework/sequential_executor.cc [hunk 3]
```diff
@@ -289,7 +289,7 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
       // call compute on the kernel
       VLOGS(logger, 1) << "Computing kernel: " << node_name_for_profiling;
 
-      kernel_begin_time = session_state.Profiler().StartTime();
+      kernel_begin_time = session_state.Profiler().Now();
```
Comment (zhanghuanrong) [inline onnxruntime/core/framework/sequential_executor.cc:292]: I know this is old code. But personally I feel this get kernel_begin_time should move down, and put it at the last step for the preparation, i.e., after CalculateTotoalInputSizes().

File: onnxruntime/core/common/profiler.h (hunk 1 above)
Comment (RandySheriffH) [reply onnxruntime/core/common/profiler.h:81]: As overriding alternatives, these method allows the consumer of profiler to log the duration of an op under different situation.

File: onnxruntime/core/framework/sequential_executor.cc [hunk 4]
```diff
@@ -357,9 +357,10 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
                 << "\n";
 #endif
 
+      kernel_end_time = session_state.Profiler().Now();
```
Comment (zhanghuanrong) [inline onnxruntime/core/framework/sequential_executor.cc:360]: Same here. We best move kernel_end_time = session_state.Profiler().Now(); at the begin() of post processing block. 
i.e., before the CalculateTotalOutputSizes @ line 345.

File: onnxruntime/core/common/profiler.h (hunk 2 above)
Comment (RandySheriffH) [reply onnxruntime/core/common/profiler.h:89]: 'dur' is the duration of the running time of an op, while 'ts' is the time difference between profiler start time and op start time.

File: onnxruntime/core/framework/sequential_executor.cc (hunk 4 above)
Comment (RandySheriffH) [reply onnxruntime/core/framework/sequential_executor.cc]: Done.

File: onnxruntime/core/framework/sequential_executor.cc (hunk 3 above)
Comment (RandySheriffH) [reply onnxruntime/core/framework/sequential_executor.cc]: Done.

File: onnxruntime/core/common/profiler.h (hunk 2 above)
Comment (pranavsharma) [reply onnxruntime/core/common/profiler.h:89]: Can you document and rename these parameters appropriately that reflects the meaning esp. now that these methods are exposed as public methods?

**Review by zhanghuanrong — APPROVED** [review]
Ship it!

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**